maplap/interface.py

```diff
@@ -0,0 +1,641 @@
+"""interface"""
+import sys
+
+from PyQt5 import Qt, QtCore, QtGui, QtWidgets
+
+import constant as C
+from design import UiMapLap
+
+
+class Rectangle:
+    """Rectangle struct"""
+
+    def __init__(self, x1=C.ZERO, y1=C.ZERO, x2=C.ZERO, y2=C.ZERO):
+        self.x_start = x1
+        self.y_start = y1
+        self.x_end = x2
+        self.y_end = y2
+
+    def none_for_pylint(self):
+        """do nothing"""
+        self.none_for_pylint()
+
+    def also_none_for_pylint(self):
+        """also do nothing"""
+        self.also_none_for_pylint()
+
+
+class SettingsParams:
+    """Params for settings"""
+
+    # pylint: disable=R1721
+    # I know that better - ok
+
+    def __init__(self, line_param):
+        params = [elem for elem in list(line_param.split(","))]
+        assert len(params) == C.PARAMS_IN_SETTING
+        for i in range(C.PARAMS_IN_SETTING):
+            setattr(self, C.SETTINGS_PARAM_ATR[i], params[i])
+
+    def none_for_pylint(self):
+        """do nothing"""
+        self.none_for_pylint()
+
+    def also_none_for_pylint(self):
+        """also do nothing"""
+        self.also_none_for_pylint()
+
+
+class Settings:
+    """Settings struct"""
+
+    # pylint: disable=R1721
+    # I know that better - ok
+
+    def __init__(self):
+        file_settings = open(C.DEFAULT_SETTINGS, "r")
+        all_lines = [line for line in file_settings]
+        assert len(all_lines) == C.SETTINGS_PARAMS
+        for i in range(C.SETTINGS_PARAMS):
+            setattr(self, C.SETTINGS_ATR[i], SettingsParams(all_lines[i]))
+        file_settings.close()
```

Comment:
В Python с файлами работают так:
```suggestion
        with open(C.DEFAULT_SETTINGS, "r") as file_settings:
            all_lines = [line for line in file_settings]
            assert len(all_lines) == C.SETTINGS_PARAMS
            for i in range(C.SETTINGS_PARAMS):
                setattr(self, C.SETTINGS_ATR[i], SettingsParams(all_lines[i]))
```
чтобы `close` вызывался автоматически даже в случае исключения.